### PR TITLE
SCMOD-7280 : Fortify fixes for Unreleased Resource : Streams

### PR DESCRIPTION
**Pull request by gaana-vishwnath**
JENKINS: http://sou-jenkins2.hpeswlab.net/job/CAFAudit/job/CAFAudit~audit-service~SCMOD-7280~CI/2/console

JIRA: https://portal.digitalsafe.net/browse/SCMOD-7280

### CODE REVIEW

File: caf-audit-monkey-container/pom.xml [hunk 1]
```diff
@@ -47,10 +47,6 @@
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-slf4j-impl</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.apache.logging.log4j</groupId>
-            <artifactId>log4j-api</artifactId>
-        </dependency>
```
Comment (dermot-hardy) [inline caf-audit-monkey-container/pom.xml:53]: Is this change for this Jira?

File: caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java [hunk 2]
```diff
@@ -70,10 +71,16 @@ public String load(String indexName) {
 
     private XContentBuilder getTenantIndexTypeMappingsBuilder() {
         //  Get the contents of the index mapping file and assign to byte array before attempting to parse JSON
-        byte[] cafAuditEventTenantIndexMappingsBytes;
-        try {
-            cafAuditEventTenantIndexMappingsBytes = ByteStreams.toByteArray(getClass().getClassLoader()
-                    .getResourceAsStream(ElasticAuditConstants.Index.TYPE_MAPPING_RESOURCE));
+        final byte[] cafAuditEventTenantIndexMappingsBytes;
```
Comment (dermot-hardy) [inline caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java]: Some rogue white-space at the end of the line.

File: caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java [hunk 3]
```diff
@@ -70,10 +71,16 @@ public String load(String indexName) {
...
-        byte[] cafAuditEventTenantIndexMappingsBytes;
-        try {
-            cafAuditEventTenantIndexMappingsBytes = ByteStreams.toByteArray(getClass().getClassLoader()
-                    .getResourceAsStream(ElasticAuditConstants.Index.TYPE_MAPPING_RESOURCE));
+        final byte[] cafAuditEventTenantIndexMappingsBytes;        
+        try (final InputStream inputStream = getClass().getClassLoader()
+                    .getResourceAsStream(ElasticAuditConstants.Index.TYPE_MAPPING_RESOURCE)){
+            if(inputStream== null)
+            {
+                String errorMessage = "Unable to read bytes from " + ElasticAuditConstants.Index.TYPE_MAPPING_RESOURCE;
```
Comment (dermot-hardy) [inline caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java]: Missing `final`.

**Review by dermot-hardy — APPROVED** [review]

File: caf-audit-monkey-container/pom.xml (hunk 1 above)
Comment (gaana-vishwnath) [reply caf-audit-monkey-container/pom.xml:53]: there was duplicity in the dependency import and the local build failed bcoz of it. So removed the duplicate entry.

**Pull request merged**